cli/azd/pkg/input/previewer.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package input
+
+import (
+	"strings"
+
+	tm "github.com/buger/goterm"
+)
+
+type previewer struct {
```

Comment:
Naming is hard. When I think about "preview" in UI terms, I think about a "preview" showing you what the output would look like if you ran something, instead of the actual output.

For example, when you hover over running applications in Windows, there's a preview popup that shows what happens when you click on the application. When you open Outlook, there's a preview pane that shows what happens if you clicked into each email.

An alternative name here that I currently have, as I read through the implementation, is an "outputFrame" or "progressFrame". The output is framed to be a specific dimension and height, with a top and a bottom.

I think @ellismg will likely have some good suggestions here as well.

Reply:
Perhaps `ProgressLog` or something along these lines.  In some ways this feels like a component that does something sort of similar to a `tail -f`.

It would be good to document the general strategy of what this thing does as well, and how you are to hold it (the fact that you write to is as an io.Writer and that ends up rendering into a subview of the current console is sort of non trivial to work out).